packages/expo-barcode-scanner/src/BarCodeScanner.tsx

```diff
@@ -0,0 +1,151 @@
+import { UnavailabilityError } from 'expo-errors';
+import mapValues from 'lodash.mapvalues';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { findNodeHandle, Platform, ViewProps, ViewPropTypes } from 'react-native';
+
+import ExpoBarCodeScannerModule from './ExpoBarCodeScannerModule';
+import ExpoBarCodeScannerView from './ExpoBarCodeScannerView';
+
+const { BarCodeType, Type } = ExpoBarCodeScannerModule;
+
+const EVENT_THROTTLE_MS = 500;
+
+type BarCodeEvent = {
+  type: string;
+  data: string;
+  [key: string]: any;
+};
+
+export type BarCodeEventCallbackArguments = {
+  nativeEvent: BarCodeEvent;
+};
+
+export type BarCodeReadCallback = (params: BarCodeEvent) => void;
+
+export type BarCodeScannedCallback = (params: BarCodeEventCallbackArguments) => void;
+
+export interface BarCodeScannerProps extends ViewProps {
+  type?: 'front' | 'back' | number;
+  torchMode?: 'on' | 'off';
+  barCodeTypes?: string[];
+  onBarCodeRead?: BarCodeReadCallback;
+  onBarCodeScanned: BarCodeScannedCallback;
+}
+
+type AnyComponent = null | React.Component<any, any> | React.ComponentClass<any>;
```

Comment:
Might be able to remove this type altogether eventually (mentioned in a comment below) for AnyComponent I think we want to include just components, not classes:
```suggestion
type AnyComponent = null | React.Component<any, any>;
```